vm/actor/src/builtin/power/policy.rs

```diff
@@ -1,107 +1,66 @@
     pub static ref CONSENSUS_MINER_MIN_POWER: StoragePower = StoragePower::from_i32(2 << 30).unwrap(); // placeholder
 
+    pub static ref BASE_MULTIPLIER: BigInt = BigInt::from(10); // PARAM_FINISH
+    pub static ref DEAL_WEIGHT_MULTIPLIER: BigInt = BigInt::from(11); // PARAM_FINISH
+    pub static ref VERIFIED_DEAL_WEIGHT_MULITPLIER: BigInt = BigInt::from(100); // PARAM_FINISH
+    pub static ref SECTOR_QUALITY_PRECISION: BigInt =  BigInt::from(20); // PARAM_FINISH
 }
 
-/// Penalty to pledge collateral for the termination of an individual sector.
-pub(super) fn pledge_penalty_for_sector_termination(
-    _pledge: &TokenAmount,
-    _term_type: SectorTermination,
-) -> TokenAmount {
-    // PARAM_FINISH
-    TokenAmount::from(0u8)
-}
-
-// Penalty to pledge collateral for repeated failure to prove storage.
-pub(super) fn pledge_penalty_for_windowed_post_failure(
-    _pledge: &TokenAmount,
-    _term_type: SectorTermination,
-) -> TokenAmount {
-    // PARAM_FINISH
-    TokenAmount::from(0u8)
-}
-
-/// Penalty to pledge collateral for a consensus fault.
-pub(super) fn pledge_penalty_for_consensus_fault(
-    pledge: TokenAmount,
-    fault_type: ConsensusFaultType,
-) -> TokenAmount {
-    // PARAM_FINISH: always penalise the entire pledge.
-    match fault_type {
-        ConsensusFaultType::DoubleForkMining => pledge,
-        ConsensusFaultType::ParentGrinding => pledge,
-        ConsensusFaultType::TimeOffsetMining => pledge,
-    }
+/// DealWeight and VerifiedDealWeight are spacetime occupied by regular deals and verified deals in a sector.
+/// Sum of DealWeight and VerifiedDealWeight should be less than or equal to total SpaceTime of a sector.
+/// Sectors full of VerifiedDeals will have a SectorQuality of VerifiedDealWeightMultiplier/BaseMultiplier.
+/// Sectors full of Deals will have a SectorQuality of DealWeightMultiplier/BaseMultiplier.
+/// Sectors with neither will have a SectorQuality of BaseMultiplier/BaseMultiplier.
+/// SectorQuality of a sector is a weighted average of multipliers based on their propotions.
+fn sector_quality_from_weight(weight: &SectorStorageWeightDesc) -> SectorQuality {
+    let sector_space_time = weight.sector_size as u64 * weight.duration;
+    let total_deal_space_time = &weight.deal_weight + &weight.verified_deal_weight;
+    assert!(BigInt::from(sector_space_time) < total_deal_space_time);
+
+    let weighted_base_space_time = (sector_space_time - total_deal_space_time) * &*BASE_MULTIPLIER;
+    let weighted_deal_space_time = &weight.deal_weight + &*DEAL_WEIGHT_MULTIPLIER;
+    let weighted_verified_space_time =
+        &weight.verified_deal_weight * &*VERIFIED_DEAL_WEIGHT_MULITPLIER;
+    let weighted_sum_space_time =
+        weighted_base_space_time + (weighted_deal_space_time + weighted_verified_space_time);
+    let scale_up_weighted_sum_space_time =
+        weighted_sum_space_time << ToPrimitive::to_usize(&*SECTOR_QUALITY_PRECISION).unwrap();
```

Comment:
this one specifically should not be a big int, that precision should just be a const usize to not deal with the ToPrimitive or unwrap